python_bindings/rig_scp.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <Python.h>
+#include <rs.h>
+
+/* SCPConnection type. */
+typedef struct {
+  PyObject_HEAD
+  rs_conn_t *connection;
+} SCPConnection;
+
+static PyObject* SCPConnection_new(PyTypeObject *type, PyObject *args,
+                                   PyObject *kwds)
+{
+  // Create a new SCPConnection
+  SCPConnection *self;
+  self = (SCPConnection *)type->tp_alloc(type, 0);
+  if(self != NULL)
+  {
+    self->connection = NULL;
+  }
+
+  return (PyObject *) self;
+}
+
+static int SCPConnection_init(SCPConnection *self, PyObject *args,
+                              PyObject *kwds)
+{
+  // Parse the arguments and keyword arguments
+  const char *hostname = NULL;  // Host to communicate with
+  unsigned int port = 17893;  // Port to communicate on
+  double timeout = 0.5;  // Duration to timeout after
+  unsigned int n_tries = 5;  // Number of retries before transmitting
+  unsigned int window_size = 8;  // Number of allowable outstanding packets
```

Comment:
This should default to 1 until the machine has been queried to discover otherwise.